### PR TITLE
Support URLs in `start` command

### PR DESCRIPTION
**Pull request by dmatos2012**
# Description
Fixes issue #7792 Fix basically by @kubouch , "stolen" by me :). Edit: Added "proper" fix rather than the one liner
```
start Cargo.toml
// Opens in default editor 
```

```
start https://www.google.com [alternative_browser]
// Opens page in default browser or another browser if specified
```

# User-Facing Changes

_(List of all changes that impact the user experience here. This helps us keep track of breaking changes.)_

# Tests + Formatting

Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- [X] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [X] `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [X] `cargo test --workspace` to check that all tests pass

# After Submitting

If your PR had any user-facing changes, update [the documentation](https://github.com/nushell/nushell.github.io) after the PR is merged, if necessary. This will help us keep the docs up to date.

### CODE REVIEW

**Comment by kubouch** [comment]
Unfortunately, I wasn't correct, the arguments also need to be passed to the program. Now, it doesn't throw an error but it still probably doesn't work properly.

**Comment by dmatos2012** [comment]
Yeah, thats true.  Is the extra argument supported? Is it meant to open a website or do other things? Because if its not something too complicated, I could take a look but not sure in which direction the team wants to take this command

**Comment by kubouch** [comment]
I think it would need to be supported by the `open` crate. Passing arguments to applications is a common functionality so chances are it supports it already. Hacking it ourselves seems like it would be unnecessarily complex and at that point, I'd rather move it away from core Nushell into a plugin, for example. I haven't looked into it personally, though.

**Comment by dmatos2012** [comment]
Actually just checked it, and its indeed supported by the `open` crate. Should not be too difficult so i will work on the proper fix on this same PR :)

**Comment by dmatos2012** [comment]
I noticed that the `open` crate supported opening with an specified program, so added that ability. Additionally, noticed before the filename was optional, but in this case it should be required so I thought it was fine to eliminate that part of the code. 

I think the oriiginal issue #7792 should be fixed now. 

Please let me know any other changes, and will get to them as soon as I am able :)

**Comment by fdncred** [comment]
Sorry for the delay. These changes seem reasonable to me. Can you please fix the conflicts?

**Comment by sholderbach** [comment]
I am under the inpression that @kubouch and @dmatos2012 talked slightly past each other on what is warranted from #7792.

I think @amstaniec within reason assumes that `start` behaves similar to `Windows-R` for which `msedge nushell.sh` is a valid expression (curiously under my default config neither pwsh nor cmd.exe can locate `msedge` directly)

@kubouch resolved that mentally that we would first resolve the first item for its type and then pass the remaining arguments as strings directly to the bound application (and in the case of Edge, an application, start the application itself and hand the arguments ahead)

@dmatos2012 interpreted it also on valid gounds that there is a need to support overriding the launched default application in some cases (e.g. for browsers or image viewer vs editor). Here the optional second argument is interpreted as the application if I groked the code correctly at a glance. This would prohibit passing further arguments or a list of files (don't know how sane that would be with other interests) to `start`.

**Comment by dmatos2012** [comment]
Yeah, that is true @sholderbach. Alternatively, maybe instead of adding an application as second argument, we could just stick to the way its now, but accept an URL as well.

As of now, `start https://www.google.com` would fail with 
```
Error: nu::shell::io_error (link)

  × I/O error
  help: Custom { kind: Other, error: "exit status: 4" }
```
because its trying to find ` start current_directory/https:/www.google.com` as a local file, but thats not the intention. 

That being said, mabye just a single URL validation(sort of what i tried to do) could improve the command, but indeed I agree that prohibiting a list of files it not the intended effect.

**Comment by sholderbach** [comment]
For overrriding the application I was thinking in the direction of having an optional flag parameter on `start` like `--app` but then we would also run into the issue that the orderring of arguments would matter to distinguish it from flags to be passed through as Jakub suggested. Here our flag parsing logic is still a source of headaches.

**Comment by kubouch** [comment]
Is overriding the default application necessary if you can simply run the application directly?

Also, the error message should have a span. Unspanned errors are strongly discouraged and shouldn't be used unless absolutely necessary, otherwise it's making scripts extremely hard to debug.

**Comment by sholderbach** [comment]
Yeah good point @kubouch!

**Comment by fdncred** [comment]
For this PR, I'd vote for this "just working" `start https://www.google.com` with the default application.  That's how it works in pwsh on Windows.

**Comment by dmatos2012** [comment]
Thanks!. I did not notice the span was missing. Also just made changes to realign the PR.

Thx!

**Comment by sholderbach** [comment]
How hard would it be to support non-prefixed urls?

**Comment by fdncred** [comment]
> How hard would it be to support non-prefixed urls?

I was thinking about this the other day too and threw the idea away because i'm not sure how we'd tell the difference between a file named `example.com` and a domain name. Maybe there's heuristic we could employ to look for common domain suffixes like .net, .org, .com, etc. but it seems a bit tricky to me.

**Comment by dmatos2012** [comment]
I agree with @fdncred , I thought about that myself but thought it would look very hacky. Alternatively, we could try always to parse as file first if it doenst start with `https`. SOmething like:
```
> start example.com
1. Open as a file. 
If present -> open (only looking for current directory)
If not present  -> try to parse as URL
 Success -> open
 Fail -> "Could not find `example.com`" error message
```
In the case of course that it starts with `https` then we treat it just as URL. 

That looks doable, what do you guys think?:)

**Comment by fdncred** [comment]
> I agree with @fdncred , I thought about that myself but thought it would look very hacky. Alternatively, we could try always to parse as file first if it doenst start with `https`. SOmething like:
> 
> ```
> > start example.com
> 1. Open as a file. 
> If present -> open (only looking for current directory)
> If not present  -> try to parse as URL
>  Success -> open
>  Fail -> "Could not find `example.com`" error message
> ```
> 
> In the case of course that it starts with `https` then we treat it just as URL.
> 
> That looks doable, what do you guys think?:)

Seems like a reasonable approach. I like only searching in the current directory for example.com for testing whether it's a file. I'd also add that if they start with http:// or https:// we can treat as a url. theoretically you could have https_somefile.txt and you wouldn't want to open that as a url.

**Comment by dmatos2012** [comment]
@fdncred I have made the changes to support `some` URLs without prefix. I added just 6 "common" extensions. Not sure if it will be nicer to have a list of 20-30 extensions but didnt know where to put it as didnt want to pollute the code with a vec of 30 elements.

Hopefully this time is ok:)

File: crates/nu-command/src/filesystem/start.rs [hunk 1]
```diff
@@ -34,45 +35,68 @@ impl Command for Start {
...
         };
-
-        let path = if let Some(path) = path {
-            path
+        let path_no_whitespace = &path.item.trim_end_matches(|x| matches!(x, '\x09'..='\x0d'));
+        // only check if file exists in current current directory
+        let file_path = Path::new(path_no_whitespace);
+        if file_path.exists() {
+            open::that(path_no_whitespace)?;
+        } else if file_path.starts_with("https://") || file_path.starts_with("https://") {
```
Comment (fdncred) [inline crates/nu-command/src/filesystem/start.rs:50]: starts_with https or starts_with https?

Comment (dmatos2012) [reply crates/nu-command/src/filesystem/start.rs]: uuupps, lol. Fixed! I meant `http://` :)

Comment (fdncred) [reply crates/nu-command/src/filesystem/start.rs]: figured that's what it was, LOL.

**Pull request merged**